enterprise/internal/db/perms_store.go

```diff
@@ -1045,23 +1020,22 @@ VALUES
 ON CONFLICT ON CONSTRAINT
   repo_permissions_perm_unique
 DO UPDATE SET
-  user_ids = excluded.user_ids,
+  user_ids = UNIQ(SORT(repo_permissions.user_ids + excluded.user_ids)),
```

Comment:
I think you can use `|` here and avoid the `UNIQ(SORT(`. Please validate.
```suggestion
  user_ids = repo_permissions.user_ids | excluded.user_ids,
```